src/execution/compiler/operator/sort_translator.cpp

```diff
@@ -10,6 +10,7 @@ namespace terrier::execution::compiler {
 SortBottomTranslator::SortBottomTranslator(const terrier::planner::OrderByPlanNode *op, CodeGen *codegen)
     : OperatorTranslator(codegen),
       op_(op),
+      use_top_k_(op->HasLimit() && op->GetOffset() == 0),
```

Comment:
1. You don't need this assumption. Just issue a top-K sort on `op->GetOffset() + op->GetLimit()` if there's any limit.
2. In general, I'm against storing redundant data. It adds complexity and is a source of inconsistency. You just need `op->HasLimit()` so you can remove this member variable.